### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by cxhandley**
Ok, so had a few issues with Heroku but got there eventually.

serene-atoll-1547.herokuapp.com

I've added a user: uname - brenton, password - melbdjango

Because I had cloned from git with the django app in a subtree I had to do:

```
 git subtree push --prefix timetracker heroku master
```

Then given the migration failed initially, I deleted migration files and did a fresh makemigrations. Given the change I then had to push with the following to avoid non-fast-forward error:

```
 git push heroku `git subtree split --prefix timetracker`:master --force
```

I know I could have done a fresh folder and done a git init but thought I'd try and keep things in the cloned to make sure I could do a simple push once I had it working. 

Also note: not sure if I needed 

```
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "timetracker.settings") 
```

before 

```
from whitenoise.django import DjangoWhiteNoise 
```

as this seemed to have thrown an error in Heroku.

### CODE REVIEW

File: timetracker/entries/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,25 +1,51 @@
 # -*- coding: utf-8 -*-
```
Comment (MarkusH) [inline timetracker/entries/migrations/0001_initial.py:1]: I don't know what exact problem you had with the migrations, but you _normally_ should not modify migration files.

File: timetracker/timetracker/settings.py [hunk 2]
```diff
@@ -20,12 +20,12 @@
 # See https://docs.djangoproject.com/en/1.8/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'fex(9l-^pfdvrzbd&&5!(f3cges$o#5o^sv=e#bbof*es$$*bl'
+SECRET_KEY = os.environ.get('DJANGO_SECRET_KEY', 'ThIs-K3Y-iSnT-a-5eCrEt')
```
Comment (MarkusH) [inline timetracker/timetracker/settings.py:23]: I'd recommend to go with `None` as a default to make sure you have one set in production.

File: timetracker/timetracker/urls.py [hunk 3]
```diff
@@ -14,9 +14,15 @@
...
 from django.contrib import admin
+from django.contrib.auth import views as auth_views
+
 from entries import urls as entry_urls
 
 urlpatterns = [
     url(r'^', include(entry_urls)),
     url(r'^admin/', include(admin.site.urls)),
+    url(r'^accounts/login/$', auth_views.login,
+    {'template_name': 'login.html'}, name='login'),
```
Comment (MarkusH) [inline timetracker/timetracker/urls.py:25]: I'd indent this row so it's easier to read, or even put it on the line before.

File: timetracker/timetracker/urls.py [hunk 4]
```diff
@@ -14,9 +14,15 @@
...
+    url(r'^accounts/logout/$', auth_views.logout,
+    {'next_page': '/'}, name='logout'),
```
Comment (MarkusH) [inline timetracker/timetracker/urls.py:27]: same here

File: timetracker/timetracker/wsgi.py [hunk 5]
```diff
@@ -13,4 +13,6 @@
 
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "timetracker.settings")
```
Comment (MarkusH) [inline timetracker/timetracker/wsgi.py:14]: Good point, this needs to stick around.

**Comment by MarkusH** [comment]
Looks really good :) :cookie: :tada:

File: timetracker/entries/migrations/0001_initial.py (hunk 1 above)
Comment (cxhandley) [inline timetracker/entries/migrations/0001_initial.py:1]: Hi Markus, I think it was when I was trying to add an author to the models. I didn't want to allow the author to be null. From the d31dd00 commit. I tried to run a manual migration but got stuck trying figure out where it was failing the migration. It asked to provide options 1 or 2 to manually set or ... (I can't recall). I looked back at previous examples but couldn't work out how this was getting stuck as there was no data in the database to get an existing user.

Comment (MarkusH) [inline timetracker/entries/migrations/0001_initial.py:1]: Correct. Django / your database needs to know to which author it should refer for existing rows. You could either set this to a hard coded user (not the best idea) or to accept that the value might be `None` for existing row, in which case you'd need to set `null=True` in your models. However, if it's `None`, your filtering on users is a bit more advanced. See the solution and the brief discussion in https://github.com/MelbDjango/lesson-eight/pull/5#discussion_r39130180

**Pull request closed without merging**